WebAPI/Controllers/RolesAndPermissionsController.cs

```diff
@@ -31,11 +84,303 @@ public RolesAndPermissionsController(IRolesAndPermissionsManager rolesAndPermiss
+        /// <response code="500">Permission Name is not correct </response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="200">Permission</response>
+        [ProducesResponseType(200, Type = typeof(Permission))]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpGet("GetPermissionByName")]
+        public IActionResult GetPermissionByName(string permissionName)
+        {
+            if (string.IsNullOrEmpty(permissionName)) return BadRequest();
+            return StatusCode(StatusCodes.Status200OK, _rolesAndPermissionsManager.GetPermissionByName(permissionName));
+        }
+
+        /// <summary>
+        ///     Insert a new permission in the DB
+        /// </summary>
+        /// <param name="permission">permission to be Inserted</param>
+        /// <response code="500">permission's data is not correct or Already Exists</response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="201">permission Created</response>
+        [ProducesResponseType(201, Type = typeof(Permission))]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpPost("InsertPermission")]
+        public IActionResult InsertPermission([FromBody] PermissionDTO permission)
+        {
+            if (permission == null) return BadRequest();
+            return StatusCode(StatusCodes.Status201Created, _rolesAndPermissionsManager.InsertPermission(permission));
+        }
+
+        /// <summary>
+        ///     removes a Permission from the DB
+        /// </summary>
+        /// <param name="permissionId">permission's Id to be deleted</param>
+        /// <response code="500"></response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="202">Permission Deleted</response>
+        [ProducesResponseType(202, Type = null)]
+        [ProducesResponseType(403, Type = null)]
+        [ProducesResponseType(500, Type = null)]
+        [HttpDelete("DeletePermission")]
+        public IActionResult DeletePermission(int permissionId)
+        {
+            if (permissionId <= 0) return BadRequest();
+            _rolesAndPermissionsManager.DeletePermission(permissionId);
+            return StatusCode(StatusCodes.Status202Accepted);
+        }
+
+        /// <summary>
+        ///     Gets the Permissions Of Role
+        /// </summary>
+        /// <remarks>
+        ///     Please Make sure to spell the rolename correctly
+        /// </remarks>
+        /// <param name="roleName">Role Name</param>
+        /// <response code="500">Role name is not correct </response>
+        /// <response code="403">UnAuthorized Access </response>
+        /// <response code="200">Role's Permission</response>
+        [ProducesResponseType(200, Type = typeof(IEnumerable<Permission>))]
```

Comment:
You are actually returning IEnumerable\<PermissionDTO\> not IEnumerable\<Permission\>